pkg/overlay/pack.go

```diff
@@ -725,7 +725,7 @@ func unpackOne(l ispec.Descriptor, ociDir string, extractDir string) error {
 			return err
 		}
 
-		err = layer.UnpackLayer(extractDir, uncompressed, nil)
+		err = layer.UnpackLayer(extractDir, uncompressed, &layer.UnpackOptions{WhiteoutMode: layer.OverlayFSWhiteout})
```

Comment:
```suggestion
                // always unpack with Overlay whiteout mode to prevent ignoring whiteouts in tar layers
                // see test/publish.bats: "building from published images with whiteouts" for more details
		err = layer.UnpackLayer(extractDir, uncompressed, &layer.UnpackOptions{WhiteoutMode: layer.OverlayFSWhiteout})
```

Reply:
Feel free to expand on the comment; I like to seem context comments when making optional choices like this one.

Reply:
Added.